examples/topics/categorical/correlograms.py

```diff
@@ -0,0 +1,68 @@
+''' A categorical plot showing the correlations in mineral content for 214 samples of glass fragments
+obtained during forensic work. The dataset contains seven variables measuring the amounts of magnesium (Mg),
+calcium (Ca), iron (Fe), potassium (K), sodium (Na), aluminum (Al), and barium (Ba) found in each glass fragment.
+The colored circles represent the correlations between pairs of these variables.
+The magnitude of each correlation is encoded in the size of the circles.
+
+.. bokeh-example-metadata::
+    :sampledata: forensic_glass
+    :apis: bokeh.plotting.figure.scatter
+    :refs: :ref:`ug_topics_categorical_scatters`
+    :keywords: scatter, correlogram
+
+'''
+import numpy as np
+import pandas as pd
+
+from bokeh.models import ColorBar, ColumnDataSource, FixedTicker, LinearColorMapper
+from bokeh.plotting import figure, show
+from bokeh.sampledata.forensic_glass import data
+from bokeh.transform import transform
+
+df = data.copy()
+
+pairs = ["Mg-Ba", "Mg-Al", "Mg-Na", "Mg-K", "Mg-Fe", "Mg-Ca", "Ca-Ba",
+         "Ca-Al", "Ca-Na", "Ca-K", "Ca-Fe", "Fe-Ba", "Fe-Al", "Fe-Na",
+         "Fe-K", "K-Ba", "K-Al", "K-Na", "Na-Ba", "Na-Al", "Al-Ba"]
+
+pair_split = [pair.split("-") for pair in pairs]
+correlations = []
+for pair in pair_split:
+    matrix = np.corrcoef(df[f"{pair[0]}"], df[f"{pair[1]}"])
+    correlation = matrix[0, 1]
+    correlations.append(correlation)
+
+new_df = pd.DataFrame({"oxide_1": [x[0] for x in pair_split], "oxide_2": [x[1] for x in pair_split],
+                       "correlation": correlations, "dot_size": [abs(corr) * 120 for corr in correlations]})
```

Comment:
That's a lot of content on single lines. It would be much more readable to change this to:
```suggestion
new_df = pd.DataFrame({
    "oxide_1": [x[0] for x in pair_split],
    "oxide_2": [x[1] for x in pair_split],
    "correlation": correlations,
    "dot_size": [abs(corr) * 120 for corr in correlations],
})
```

Reply:
"dot_size" could probably omitted and computed in the browser with a `CustomJSTransform` which would be a nice side-demonstration.

@mattpap I am assuming we don't have some other simpler way these days to make JS expressions for things like that